cid/common.py

```diff
@@ -1174,13 +1229,49 @@ def create_or_update_dataset(self, dataset_definition: dict, dataset_id: str=Non
 
         found_dataset = self.qs.describe_dataset(compiled_dataset.get('DataSetId'))
         if isinstance(found_dataset, Dataset):
+            update_dataset = False
             if update:
-                self.qs.update_dataset(compiled_dataset)
+                update_dataset = True
             elif found_dataset.name != compiled_dataset.get('Name'):
                 print(f"Dataset found with name {found_dataset.name}, but {compiled_dataset.get('Name')} expected. Updating.")
+                update_dataset = True
+            if update_dataset and get_parameters().get('on-drift', 'show').lower() != 'override' and isatty() and not cur_required:
+                while True:
+                    diff = self.qs.dataset_diff(found_dataset.raw, compiled_dataset)
+                    if diff and diff['diff']:
```

Comment:
Shouldn't this be part of helper code to keep common.py more high level?

Reply:
I would love to, but there are too many code related to user interaction and to logic. i do not see how it can be pushed outside without raising the complexity.